python/ray/data/tests/test_batch_mapper.py

```diff
@@ -0,0 +1,133 @@
+import numpy as np
+import pandas as pd
+import pyarrow as pa
+
+import pytest
+from typing import Dict
+from pandas.testing import assert_frame_equal
+
+import ray
+from ray.data.preprocessors import BatchMapper
+
+
+@pytest.fixture
+def test_ds_pandas_format():
+    old_column = [1, 2, 3, 4]
+    to_be_modified = [1, -1, 1, -1]
+    in_df = pd.DataFrame.from_dict(
+        {"old_column": old_column, "to_be_modified": to_be_modified}
+    )
+    ds = ray.data.from_pandas(in_df)
+    yield ds, old_column
+
+
+@pytest.fixture
+def test_ds_multi_column_arrow_format():
+    old_column = [1, 2, 3, 4]
+    to_be_modified = [1, -1, 1, -1]
+    ds = ray.data.from_arrow(
+        pa.table(
+            {
+                "old_column": old_column,
+                "to_be_modified": to_be_modified,
+            }
+        )
+    )
+    yield ds, old_column
+
+
+def test_batch_mapper_pandas_data_format(test_ds_pandas_format):
+    """Tests batch mapper functionality."""
+    ds, old_column = test_ds_pandas_format
+
+    def add_and_modify_udf_pandas(df: "pd.DataFrame"):
+        df["new_col"] = df["old_column"] + 1
+        df["to_be_modified"] *= 2
+        return df
+
+    def add_and_modify_udf_numpy(data: Dict[str, np.ndarray]):
+        data["new_col"] = data["old_column"] + 1
+        data["to_be_modified"] *= 2
+        return data
+
+    expected_df = pd.DataFrame.from_dict(
```

Comment:
Nit: base DataFrame constructor works here!

```suggestion
    expected_df = pd.DataFrame(
```